### DISCUSSION TITLE
plagiarism

### DISCUSSION THREAD

**Discussion by smriti1313** [TesseractCoding/NeoAlgo#1704] [Q&A] [answered]
Hello, while reading the main README, it has been mentioned that no plagiarism will be entertained. I know that plagiarism means taking work of someone else. But if we were to provide solution to algorithms then it might be possible that it will match to someone's code on internet unless we change the variable names . I am not getting that plagiarism point clearly. Kindly help me.
Thank you

**Comment by vatsalkesarwani12** [comment, marked as answer]
Hello @smriti1313 
When we say about plagiarism we see that you haven't copied the solution from any site.
We see various points like a variable, comments, description, sometimes code styles.
Also when there is asked to follow a proper guideline for submitting a solution and the solution doesn't stand the guidelines we add a plag review tag and check a few sites online. And if you are caught we mark you as plag. 
So that's why we add contributors., etc guidelines for the participants to follow and submit pr according to them.

Hope you get your answer.

**Comment by HarshCasper** [comment]
Hi @smriti1313 

@plazzy99 has given a detailed answer, and I would like to add just a few points here: 

- Plagiarism would count if the whole Code is copied with no change whatsoever. 
- You can take inspiration from some Sites, maybe pick up some Boilerplate Code but the core implementation should be yours.
- During the review process we might ask to make multiple changes. That affirms if your code is plagiarized or not.

The Plagiarism Review is subjective. It might be possible that we would not be able to identify it in plain sight. But to ensure that the sanity of the Codebase is maintained over a period, we will try to strictly review your Code, so that plagiarized PRs don't find their way through. 

I hope that answers your question 👍
